src/AccountIdEndpointMiddleware.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Aws;
+
+use Aws\Exception\CredentialsException;
+
+/**
+ * This middleware class resolves the identity from a credentials provider callable function
+ * and determine whether an account should have been resolved. When this middleware resolves
+ * identity then, the identity is included in the $command context bag property "$command['@context]"
+ * as resolved_identity. Example $command['@context]['resolved_identity'] = $resolvedIdentity, and
+ * when this property is set then, the signer middleware gives preference to use that resolved identity
+ * instead of resolving the provided credentials provider by the client. This is done to avoid having to
+ * resolve credentials more than once per request.
+ */
+class AccountIdEndpointMiddleware
+{
+    /**
+     * @var callable $nextHandler
+     */
+    private $nextHandler;
+    /**
+     * @var string $accountIdEndpointMode
+     */
+    private $accountIdEndpointMode;
+    /**
+     * @var callable $credentialsProvider
+     */
+    private $credentialsProvider;
+
+    /**
+     * @param callable $nextHandler
+     * @param string $accountIdEndpointMode
+     * @param callable $credentialsProvider
+     */
+    public function __construct($nextHandler, $accountIdEndpointMode, $credentialsProvider)
```

Comment:
Might want to consider passing an array of args/config here and pulling the endpoint mode out of that.  Never know what other values we may need access to in the future

Reply:
Middleware has been removed.